### ISSUE TITLE
Rednose makes Py2.6 builds fail

### ISSUE THREAD

**Issue by wojcikstefan** [JBKahn/rednose#9]
Hi there,

I just noticed while doing some work on https://github.com/MongoEngine/mongoengine/ that their Py2.6 tests fail and the issue can be traced back to `rednose` (see the traceback at the bottom of this issue). Removing your package fixes the issue.

For debugging, compare a failed build (e.g. https://travis-ci.org/MongoEngine/mongoengine/jobs/167092804) with a fixed build (e.g. https://travis-ci.org/MongoEngine/mongoengine/jobs/169038786).

Unfortunately, I don't know enough about this package to help.

```
======================================================================
1) ERROR: test_binary_field_primary_filter_by_binary_pk_as_str (tests.FieldTest)
----------------------------------------------------------------------
   Traceback (most recent call last):
    .tox/py26-mg27/lib/python2.6/site-packages/nose/case.py line 133 in run
      self.runTest(result)
    .tox/py26-mg27/lib/python2.6/site-packages/nose/case.py line 151 in runTest
      test(result)
    .tox/py26-mg27/lib/python2.6/site-packages/nose/proxy.py line 132 in addError
      self.result.addError(self.test, self._prepareErr(err))
    .tox/py26-mg27/lib/python2.6/site-packages/rednose.py line 261 in addError
      self.addSkip(test, err)
    .tox/py26-mg27/lib/python2.6/site-packages/rednose.py line 275 in addSkip
      skip_message = "#{test_id} {test_location} ... ".format(test_id=self._get_id(test), test_location=test.context.__file__)
   AttributeError: type object 'FieldTest' has no attribute '__file__'
```

**Comment by JBKahn** [comment]
This repo doesn't officially support python 2.6, I can look into this when I have time but can you check if this is reproducable witha  unit test. If you can provide a single failing unit test I can try to work on it.

**Comment by LordGaav** [comment]
I can reproduce this on any version of Python when raising a SkipTest exception (imported as `from nose import SkipTest`) and verbosity=2.

Reproducable with this code:

```python
import nose
import unittest


class TestBug(unittest.TestCase):
    @classmethod
    def setUpClass(cls):
        raise nose.SkipTest("SKIPPING!")

    def test_bug(self):
        pass
```

And this command line: `nosetests --rednose --verbosity=2`

Results in:

```
Traceback (most recent call last):
  File "XXX/vendor/bin/nosetests", line 11, in <module>
    sys.exit(run_exit())
  File "XXX/vendor/lib/python3.4/site-packages/nose/core.py", line 121, in __init__
    **extra_args)
  File "/usr/lib/python3.4/unittest/main.py", line 93, in __init__
    self.runTests()
  File "XXX/vendor/lib/python3.4/site-packages/nose/core.py", line 207, in runTests
    result = self.testRunner.run(self.test)
  File "XXX/vendor/lib/python3.4/site-packages/nose/core.py", line 62, in run
    test(result)
  File "XXX/vendor/lib/python3.4/site-packages/nose/suite.py", line 178, in __call__
    return self.run(*arg, **kw)
  File "XXX/vendor/lib/python3.4/site-packages/nose/suite.py", line 225, in run
    test(orig)
  File "/usr/lib/python3.4/unittest/suite.py", line 84, in __call__
    return self.run(*args, **kwds)
  File "XXX/vendor/lib/python3.4/site-packages/nose/suite.py", line 76, in run
    test(result)
  File "XXX/vendor/lib/python3.4/site-packages/nose/suite.py", line 178, in __call__
    return self.run(*arg, **kw)
  File "XXX/vendor/lib/python3.4/site-packages/nose/suite.py", line 225, in run
    test(orig)
  File "XXX/vendor/lib/python3.4/site-packages/nose/suite.py", line 178, in __call__
    return self.run(*arg, **kw)
  File "XXX/vendor/lib/python3.4/site-packages/nose/suite.py", line 215, in run
    result.addError(self, self._exc_info())
  File "XXX/vendor/lib/python3.4/site-packages/nose/proxy.py", line 132, in addError
    self.result.addError(self.test, self._prepareErr(err))
  File "XXX/vendor/lib/python3.4/site-packages/rednose.py", line 261, in addError
    self.addSkip(test, err)
  File "XXX/vendor/lib/python3.4/site-packages/rednose.py", line 275, in addSkip
    skip_message = "#{test_id} {test_location} ... ".format(test_id=self._get_id(test), test_location=test.context.__file__)
AttributeError: type object 'TestBug' has no attribute '__file__'
```

**Comment by JBKahn** [comment]
Awesome, I'll try to take a look at this repo this week and get a fix in.

**Comment by miki725** [comment]
can confirm issue can be replicated in py2.7

**Comment by JBKahn** [comment]
fixed and shipped 1.2.2
